File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/Searching.cpp
#include "Searching.h"
#include "IInventoryInterface.h"
#include <vector>
#include <string>

namespace Order
{
  Searching::Searching()
  {
  }
  int Searching::searchProduct(std::string productName)
  {
    std::vector<int> productIds = this->dependency_SearchInventory->getProductIds();
    for (auto& productId : productIds)
    {
      std::string name = this->dependency_SearchInventory->getProductName(productId);
      if (name == productName)
      {
        return productId;
      }
    }
    return -1;
  }
}

File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/Searching.cs
/// <filename>
///     Searching.cs
/// </filename>
using Order;
using System.Collections.Generic;

namespace Order
{
    public class Searching : Order.ISearchInterface
    {

        public Searching()
        {
        }

        public virtual int searchProduct(string productName)
        {
            System.Collections.Generic.List<int> productIds = this.dependency_SearchInventory.getProductIds();
            foreach (var productId in productIds)
            {
                string name = this.dependency_SearchInventory.getProductName(productId);
                if (name == productName)
                {
                    return productId;
                }
            }
            return -1;
        }

        public Order.IInventoryInterface dependency_SearchInventory;
    }
}
